### ISSUE TITLE
[BUG]: Ascend950 v3/v4 FD returns incorrect O/LSE due to missing synchronization before LM buffer reuse

### ISSUE THREAD

**Issue by 780926832** [MinghuasLab/flash-attention-npu#294]
### Preflight checklist

- [x] I have searched the existing issues and confirmed this is not a duplicate.
- [x] I can reproduce this on the latest release.
- [x] I have removed any sensitive information from logs or code snippets.

### Environment

- Device: Ascend950PR_9579
- Actual AIC count: 28
- Host architecture: x86_64
- CANN: 9.2.0.beta2
- Python: 3.12.14
- PyTorch: 2.12.0+cpu
- torch_npu: 2.12.0
- ATK: 26.7.8.dev20260730

Tested source revisions:
- v3 baseline: 606b919, before applying the v3 LM synchronization fix.
  This revision already contains the v4 fix.
- v4 baseline: 4e9d854, without the LM synchronization fix.
  The relevant v4 forward code was also checked against upstream/main
  at cab322e and was unchanged.

The tests execute real Ascend950 kernels.

### Bug Description

Ascend950 v3/v4 的 Flash Decoding 路径存在共享 LM 缓冲区的同步问题，
可能导致输出 O 和 softmax LSE 同时出现精度错误。

RescaleO 将 partial LSE 写入 LM，然后通过 MTE3 异步搬运到全局内存。
下一基础任务的 OnlineSoftmax 会复用同一块 LM。如果此前的搬运尚未
完成，MTE3 就可能读到已被覆盖的数据。

原有 MTE3_V EVENT_ID4 等待位于下一次 RescaleO/SubCoreCompute 入口，
但下一任务的 Softmax 可以先执行，因此该等待无法保护 LM 的复用。

问题与执行时序有关，同一用例单次通过不能排除该问题。
FP16、BF16 均已观察到精度失败。

Expected:
O 和 LSE 应满足现有 ATK 三方精度标准。

Actual:
算子正常执行结束，但 O/LSE 精度检查失败。

### Steps to Reproduce

1. 使用未包含对应 LM 同步修复的源码，加载 CANN 环境，
   编译安装 Ascend950 v3：

   FLASH_ATTN_FORCE_REBUILD=TRUE \
   FLASH_ATTN_BUILD_VERSION=v3 \
   FLASH_ATTN_BUILD_NPU=950 \
   python setup.py install

2. 确认 ATK 所用 Python 加载的是刚编译的未修复扩展，
   没有通过 PYTHONPATH 加载其他修复版本。

3. 在 ATK 的以下工程中使用随 issue 提供的复现 JSON：
   interfaces/ascend950/flash_attn_with_kvcache_v3

   将 nodes_accuracy.yaml 配置到实际具有 28 个 AIC 的 Ascend950 设备。

4. 执行代表用例：

   FA_ATK_AIC_NUM=28 OMP_NUM_THREADS=4 atk task \
     -c repro_fd_lm_sync_28core.json \
     -n nodes_accuracy.yaml \
     -p execute_flash_attn_with_kvcache_v3.py \
     --task accuracy \
     --bm_device cpu \
     --single_process \
     --white_list '[820004]' \
     --save_data input \
     --save_data output

5. 检查报告中的 O/LSE 精度结果，不仅检查 ATK 的 SUCCESS 状态。
   如单次通过，可独立重复执行同一命令，保持原始 seed。

### Error output

```shell
| 参数 | 值 |
|---|---|
| v3 case ID | `820004` |
| seed | `20361678` |
| dtype | FP16 |
| B / Sq / Hq / Hkv / D / Sk | `1 / 13 / 24 / 4 / 192 / 1024` |
| Q shape | `[13,24,192]` |
| K/V cache shape | `[8,128,4,192]` |
| page size | `128` |
| causal | `False` |
| window_size | `(-1,-1)` |
| num_splits | `8` |
| scheduler_metadata | `None`，Host FD |
| return_softmax_lse | `True` |
没有运行时异常，以下为 v3 case 820004 的 ATK 精度报告摘录。

O:
Actual RMSE = 0.002168698236346245
ERROR: 均方根误差比例(35.531951904296875)超过阈值(1.2)

LSE:
Actual RMSE = 0.04201843589544296
ERROR: 均方根误差比例(688.4300537109375)超过阈值(1.2)

该轮未修复 v3 对照测试：
24 个用例中，12 个通过、12 个精度失败，0 个执行异常。
12 个失败用例的 O 和 LSE 均未通过。
```
